### DISCUSSION TITLE
How can I make dyn4j FPS independant?

### DISCUSSION THREAD

**Discussion by ia97lies** [dyn4j/dyn4j#285] [General]
Hi,
I have some issues with FPS, I don't know how to handle that properly. I tried to set the time per frame on every update, so it would be FPS independent. What I saw is that if I put a higher FPS than 60 the applied force has a much less impact, not sure what I did wrong, but it needs then more force?! Any idea what the issue could be? It works ok (not great) if FPS drops to 30 (with step frequency 1/60d)
I set the FPS with the settings like this:
        settings.setStepFrequency(tpf);

Do I miss something?

**Comment by wnbittle** [comment]
When you say you need to apply more force to compensate for a drop in FPS, could you share the code where you are applying the force?  Also, could you share the code where to are telling the engine to "run" or perform an update?

**Reply by ia97lies** [reply]
A bit late but here is the code:
 
  @Override
    public void updateWithPause(double tpf) {
        currentTpf = tpf;

        rectangleBodyContainer.update();
        sphereBodyContainer.update();
        boarderContainer.update();
        namedBorderContainer.update();


        running = frameRateStabilizer.checkFrameRateStability(currentTpf, lastTpf);

        if (running) {
            forceContainer.update();
            torqueContainer.update();
            velocityContainer.update();
            applyForces();
            applyTorques();
            applyVelocity();
            this.world.update(currentTpf);
        }

        propagateNewPositions();
        lastTpf = currentTpf;
        currentTpf = 0;
    }


    private void applyForces() {
        forcesByParent.entrySet().stream()
                .filter(parent -> getBody(parent.getKey()) != null)
                .forEach(parent -> {
                    Dyn4jBody body = getBody(parent.getKey());
                    parent.getValue().values().forEach(force -> body.applyImpulse(new Vector2(force.x, force.y)));
                    parent.getValue().keySet().forEach(entityId -> ed.removeEntity(entityId));
                });
    }


I start to see the problem, it is an impulse not a force I apply.
